### ISSUE TITLE
Feature request: generate top k sequences

### ISSUE THREAD

**Issue by yzhang-github-pub** [lucidrains/x-transformers#133]
Instead of generate the best (top 1) sequence from autoregressive generate function, it is useful to generate >1 best sequences. For example, in language translation, human users can have choices from more than 1 best translations.

Thanks for your consideration.

**Comment by yzhang-github-pub** [comment]
@lucidrains 
Not exactly. Using language translation as an example. The model generates one best target sentence,  either greedily or by beam search. What I ask is to generate top k target sentences.

**Comment by maryszmary** [comment]
For me, it would actually be great if `generate` had a beam search option, since I need multiple outputs ranked by probability for character-level generation of phonetic transcriptions.

**Comment by lucidrains** [comment]
@maryszmary yea i should add classic beam search at some point for sure

maybe roll up any interesting advancements in that area of research with its release? lemme know
